### PR TITLE
Do not export internal symbols in shared object files

### PR DESCRIPTION
**Pull request by jrohel**
Closes https://github.com/rpm-software-management/dnf5/issues/1304

PR disables C++ unit tests for libdnf5, libdnf5-cli, dnf5 cpr_plugin C++. These unit tests use private symbols that are no longer accesible. It is a temporarily solution until the unit tests are modified.

### CODE REVIEW

File: include/libdnf5/defs.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF5_DEFS_H
+#define LIBDNF5_DEFS_H
+
+#define PUBLIC_API __attribute__((visibility("default")))
```
Comment (ppisar) [inline include/libdnf5/defs.h:23]: __attribute__((visibility())) is a compiler-specific extension. E.g. GCC and Clang have them. Other compilers might not have. Consider guarding the definition with __GNUC__ preprocessor condition.

File: bindings/libdnf5/advisory.i [hunk 2]
```diff
@@ -38,6 +38,9 @@
 
 #define CV __perl_CV
 
+#define LIBDNF_API
+#define LIBDNF_LOCAL
```
Comment (jan-kolarik) [inline bindings/libdnf5/advisory.i:42]: Can we define it once, e.g. in `shared.i`?

File: dnf5/include/dnf5/context.hpp [hunk 3]
```diff
@@ -249,25 +250,25 @@ class RpmTransCB : public libdnf5::rpm::TransactionCallbacks {
     void verify_stop([[maybe_unused]] uint64_t total) override;
 
 private:
-    void new_progress_bar(int64_t total, const std::string & descr);
+    DNF_LOCAL void new_progress_bar(int64_t total, const std::string & descr);
```
[the remark this answers is not in the document]
Comment (jrohel) [reply dnf5/include/dnf5/context.hpp:253]: For private classes, the symbols are hidden by default. But this is a public API class. It is defined by `class DNF_API class_name`. Such a class has symbols public (exported) by default. That's why I explicitly set DNF_LOCAL for private methods.

File: bindings/libdnf5/advisory.i (hunk 2 above)
Comment (jrohel) [reply bindings/libdnf5/advisory.i]: OK. I moved the macro definition to `shared.i`.

**Comment by jan-kolarik** [comment]
Thanks, looks good. Just a rebase is probably needed to fix tests.

**Comment by jrohel** [comment]
The Locker class has been moved to the public API in upstream.
So, I did a rebase and modification of this PR.

**Comment by jan-kolarik** [comment]
The current tests issues are connected with the new rpm `4.19.91-6` in Rawhide. New `4.19.91-7` version should fix the issues (tested locally). Waiting to get into compose and we can re-run the tests...

**Comment by jan-kolarik** [comment]
Sorry about that, but could you rebase again please? From the other PRs, it seems that the CI is finally working again...

**Pull request merged**